firedrake/assemble.py

```diff
@@ -355,6 +372,7 @@ def _make_tensor(form, bcs, *, diagonal, mat_type, sub_mat_type, appctx,
         raise AssertionError
 
 
+# FIXME: This class is not annotated for the adjoint so taping will not work as expected.
```

Comment:
This FIXME probably needs addressing before merge

Reply:
This is essentially just a documentation fix to go with https://github.com/firedrakeproject/firedrake/issues/2581. I thought it would do no harm to merge alongside this PR.